### PR TITLE
Improve no result found sentence in doc search

### LINKED ISSUE #55531: Add more information in case of search failure
Like suggesting links to books (both rust and rustdoc books?).

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #55531.

Look:

screenshot 2018-11-01 at 02 21 56

r? @QuietMisdreavus

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -1340,7 +1340,13 @@
                 output = '<div class="search-failed"' + extraStyle + '>No results :(<br/>' +
                     'Try on <a href="https://duckduckgo.com/?q=' +
                     encodeURIComponent('rust ' + query.query) +
-                    '">DuckDuckGo</a>?</div>';
+                    '">DuckDuckGo</a>?<br/><br/>' +
+                    'Or maybe you want to take a look to the <a ' +
```
Comment (varkor) [inline src/librustdoc/html/static/main.js]: "Take a look at" or "Look in" sounds more natural here (and in "or to" below).

**Comment by varkor** [comment]
Maybe it's worth being even more explicit:
```
Or try looking in one of these:
- The [Rust Reference] for technical details about the language.
- [Rust by Example] for expository code examples.
- The [Rust Book] for introductions to language features and the language itself.
```
Just in case the reader isn't familiar with them already (there's enough spare room).

File: src/librustdoc/html/static/main.js (hunk 1 above)
Comment (memoryruins) [reply src/librustdoc/html/static/main.js]: When doc searches are in other crates and not std, the "or maybe you want" might not apply as often. The following is a plain option that links to the full Rust Documentation page for providing descriptions and additional resources:

> [Rust Documentation](https://www.rust-lang.org/en-US/documentation.html) provides additional resources and guides, such as the [Rust Book](https://doc.rust-lang.org/book/index.html), [Rust by Example](https://doc.rust-lang.org/rust-by-example/), and the [Rust Reference](https://doc.rust-lang.org/stable/reference/).

**Comment by QuietMisdreavus** [comment]
If this is going to be in every doc build and not just std (as it is right now), then i'm wary about this. Imagine you're looking for something in hyper's docs and get linked to std. If anything, we could link to all of these things, I like @varkor's suggestion, if we also include a link to docs.rs, such as:

```markdown
- [Docs.rs] for documentation on third-party crates.
```

**Comment by GuillaumeGomez** [comment]
Way better indeed! I'll use @varkor's suggestion.

**Comment by GuillaumeGomez** [comment]
New look:

screenshot 2018-11-04 at 13 04 56

**Comment by QuietMisdreavus** [comment]
Looks good to me! @varkor @memoryruins, what do y'all think?

**Comment by memoryruins** [comment]
I like it :) the links will be useful!

Since the new list looks best left-aligned yet is logically grouped with the "result not found" text, thoughts on left-aligning everything altogether?

no_results

The left side is where my eyes already go when expecting a result.

Instead of using the term "third-party" on the Docs.rs description, should we say 
```
- [Docs.rs] for documentation of crates released on crates.io
```

Some rust-nursery crates are nearly std and it could be helpful to specify the origin of the generated docs.

**Comment by QuietMisdreavus** [comment]
+1 to @memoryruins' wording suggestion.

**Comment by GuillaumeGomez** [comment]
Updated the wording.

**Review by QuietMisdreavus — APPROVED** [review]
Thanks so much! This looks good to me.

**Pull request merged**